# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1328632893 18000
#      Tue Feb 07 11:41:33 2012 -0500
# Branch stable
# Node ID 2cd56a5e3a66f70336e91269adf311d6606a5a68
# Parent  71efccec5936686eaf08af4dab302a7f921d736e
new tests

* ols.m: New tests.

diff --git a/scripts/statistics/base/ols.m b/scripts/statistics/base/ols.m
--- a/scripts/statistics/base/ols.m
+++ b/scripts/statistics/base/ols.m
@@ -136,16 +136,37 @@ endfunction
 
 
 %!test
 %! x = [1:5]';
 %! y = 3*x + 2;
 %! x = [x, ones(5,1)];
 %! assert (ols(y,x), [3; 2], 50*eps)
 
+%!test
+%! x = [1, 2; 3, 4];
+%! y = [1; 2];
+%! [b, s, r] = ols (x, y);
+%! assert (b, [1.4, 2], 2*eps);
+%! assert (s, [0.2, 0; 0, 0], 2*eps);
+%! assert (r, [-0.4, 0; 0.2, 0], 2*eps);
+
+%!test
+%! x = [1, 2; 3, 4];
+%! y = [1; 2];
+%! [b, s] = ols (x, y);
+%! assert (b, [1.4, 2], 2*eps);
+%! assert (s, [0.2, 0; 0, 0], 2*eps);
+
+%!test
+%! x = [1, 2; 3, 4];
+%! y = [1; 2];
+%! b = ols (x, y);
+%! assert (b, [1.4, 2], 2*eps);
+
 %% Test input validation
 %!error ols ();
 %!error ols (1);
 %!error ols (1, 2, 3);
 %!error ols ([true, true], [1, 2]);
 %!error ols ([1, 2], [true, true]);
 %!error ols (ones (2,2,2), ones (2,2));
 %!error ols (ones (2,2), ones (2,2,2));
